rabbitmq/assets/configuration/spec.yaml

```diff
@@ -7,160 +7,169 @@ files:
-        items:
-          type: string
-        example:
-          - <EXCHANGE_1>
-          - <EXCHANGE_2>
-    - name: exchanges_regexes
-      description: |
-        Use the `exchanges_regexes` parameters to specify one or multiple regex to to match
-        the exchanges you'd like to collect metrics on (up to 50 exchanges).
-        If you have less than 50 exchanges, you don't have to set this parameter,
-        the metrics is collected on all the exchanges by default.
-      value:
-        type: array
-        items:
-          type: string
-        example:
-          - <REGEX>
-    - name: vhosts
-      description: |
-        Service checks and `rabbitmq.connections` metric:
-        By default a list of all vhosts is fetched and each one is checked using the aliveness
-        API and the metric `rabbitmq.connections` is collected.
-        If you prefer only certain vhosts to be monitored with service checks
-        and the number of `connections` then you can list the vhosts you care about in this parameter.
-      value:
-        type: array
-        items:
-          type: string
-        example:
-          - <VHOST_NAME_1>
-          - <VHOST_NAME_2>
-    - name: prometheus_plugin
-      description: |
-        Settings for talking to the Prometheus plugin of RabbitMQ.
-        Specify this section to trigger metric collection from the Prometheus plugin endpoints.
+    - name: OpenMetrics
+      description: RabbitMQ Prometheus Plugin Example
       options:
-        - name: url
-          required: true
-          description: |
-            Base URL for OpenMetrics endpoints.
-          value:
+      - name: prometheus_plugin
+        description: |
+          Settings for talking to the Prometheus plugin of RabbitMQ.
+          Specify this section to trigger metric collection from the Prometheus plugin endpoints.
+        enabled: true
+        options:
+          - name: url
+            enabled: true
+            required: true
+            description: |
+              Base URL for OpenMetrics endpoints.
+            value:
+              type: string
+              example: 'http://localhost:15692'
+          - name: include_aggregated_endpoint
+            description: |
+              Whether or not to include metrics from the aggregated endpoint (/metrics).
+            value:
+              type: boolean
+              example: true
+          - name: unaggregated_endpoint
+            description: |
+              Which  unaggregated endpoint to use.
+              The choices are:
+                - If the field is left unspecified, no endpoint is used.
+                - "per-object" means the /metrics/per-object endpoint is used.
+                - "detailed" means the /metrics/detailed endpoint is used.
+                  You can also provide a query for this option, exactly as if you were making an HTTP request
+                  to this endpoint. For example, `detailed?family=queue_consumer_count`.
```

Comment:
Added an example for the detailed endpoint query

Reply:
🔥